src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_image_builder_commands.py

```diff
@@ -0,0 +1,467 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+from knack.util import CLIError
+
+from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer, live_only)
+
+from msrestazure.tools import parse_resource_id
+
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+
+# todo: store a local copy of this.
+TEST_SHELL_SCRIPT_URL = "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/quickquickstarts/customizeScript.sh"
+TEST_PWSH_SCRIPT_URL = "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/testPsScript.ps1"
+TEST_PWSH_SCRIPT_INLINE = [
+    '"mkdir c:\\buildActions"',
+    '"echo Azure-Image-Builder-Was-Here  > c:\\buildActions\\buildActionsOutput.txt"'
+]
+
+LINUX_IMAGE_SOURCE = "Canonical:UbuntuServer:18.04-LTS:18.04.201808140"
+WIN_IMAGE_SOURCE = "MicrosoftWindowsServer:WindowsServer:2019-Datacenter:2019.0.20190214"
+INDEX_FILE = "https://raw.githubusercontent.com/danielsollondon/azvmimagebuilder/master/quickquickstarts/exampleArtifacts/buildArtifacts/index.html"
+
+
+class ImageTemplateTest(ScenarioTest):
+    def _assign_ib_permissions(self, rg):   # need to manually give IB service permission to add image to grou
+        subscription_id = self.get_subscription_id()
+        assignee = "cf32a0cc-373c-47c9-9156-0db11f6a6dfc"
+
+        self.kwargs.update({
+            'sub': subscription_id,
+            'assignee': assignee,
+            'role': 'Contributor',
+            'scope': '/subscriptions/{}/resourceGroups/{}'.format(subscription_id, rg)
+        })
+
+        try:
+            self.cmd("role assignment create --assignee {assignee} --role {role} --scope {scope}")
+        except AssertionError as ex:
+            if self.is_live:
+                raise ex
+            pass
+
+    # Test framework has problem, hence, live only.
+    @ResourceGroupPreparer(name_prefix='cli_test_image_builder_template_file_')
+    @live_only()
+    def test_image_builder_template_file(self, resource_group):
+        self._assign_ib_permissions(resource_group)
+
+        self.kwargs.update({
+            'tmp': 'tmp1'
+        })
+
+        self.cmd('image builder create -g {rg} -n {tmp} --image-template "https://raw.githubusercontent.com/qwordy/azvmimagebuilder/master/quickquickstarts/0_Creating_a_Custom_Linux_Managed_Image/example.json"',
```

Comment:
seems there is no test case for local FILE ?

And I'm wondering, as this is a new test file, we didn't have any test for previous `image template` before?

Reply:
Fixed.
I add a parameter --image-template. `image template` is buried in the history. We only have `image builder` now.